feat(cae): add new datasource to get list of environments (#4551)
huaweicloud/terraform-provider-huaweicloud: docs/data-sources/cae.environments.md

---

> subcategory: "Direct Connect (CAE)"

Reviewer: ```suggestion
subcategory: "Cloud Application Engine (CAE)"
```

Author: done

---

# huaweicloud_cae_environments

Use this data source to get the list of CAE environments.

## Example Usage

```hcl
data "huaweicloud_cae_environments" "test" {
  name = "test"
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String) Specifies the region in which to query the data source.
  If omitted, the provider-level region will be used.

* `environment_id` - (Optional, String) Specifies the ID of the environment.

* `name` - (Optional, String) Specifies the name of the environment.

* `status` - (Optional, String) Specifies the status of the environment.

* `enterprise_project_id` - (Optional, String) Indicates the ID of the enterprise project that the environment belongs to.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The data source ID.

* `environment_items` - Indicates the environments list.
  The [environment_items](#CAE_environment_items) structure is documented below.

<a name="CAE_environment_items"></a>
The `environment_items` block supports:

> * `id` - The ID of the environment item.

Reviewer: ```suggestion
* `id` - The ID of the environment.
```

Author: done

* `name` - The name of the environment item.

> * `status` - The status of the environment item, enum: `creating`、`finish`、`deleting`、`freeze`、`police_freeze`、`delete_failed`.

Reviewer: ```suggestion
* `status` - The status of the environment.
```
please take these values into the description of the argument `status`, and do not contain the Chinese characters.
```
The valid values are **finish**, **freeze** and **police_freeze**.
```

Author: done

Reviewer: check your description again!

Author: done

* `annotations` - The additional attributes of the environment item.

* `created_at` - The created time of the environment item.

> * `updated_at` - The updated time of the environment item.

Reviewer: ```suggestion
* `created_at` - The creation time of the environment.

* `updated_at` - The latest update time of the environment.
```

Author: done
